daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/Typing.scala

```diff
@@ -16,6 +16,49 @@ import scala.annotation.tailrec
 
 private[validation] object Typing {
 
+  def typeOf(env: Env, exp: Expr): Type = { // testing entry point
```

Comment:
We modify slightly the testing entry point to be this top level function.

Reply:
`typeOf` is defined here because it's implementation uses the private functions `runWork` and `typeOfExpr`

Reply:
In which case, would there be any value in changing these functions to be package private rather than simply private?

If this is not possible, it is possible to use JVM reflection to access private methods (though this not ideal, it may still be better than having test code in main?)?

Reply:
I don't really see any benefit in this.
If the concern is simply that this is _testing_ only code, and so shouldn't be in main... 
we could very easily rejig so there is an an internal use also (in `checkTopExpr`)

Reply:
I changed the entry point to be `env.typeOfTopExpr` (in `Env`) and used internally.
The entry point was originally within `Env`, so the code is closer to how it was originally.

It can't be named `typeOf` because that name is used internally, and overload wont work here.

Reply:
Conclusion of offline discussion:

I'm not a huge fan of the entry points for the `Typing.scala` code being methods of `Env`. But that is the existing style used: for `checkKind`, `kindOf`, and again now for `typeOfTopExpr` (was previously called `typeOf` before this PR).

In each case, the same code is used both as a testing entry point, and within the code... but that's a good thing. It means we more closely test the same code which is being run. So I am fairly happy with the current/final state.